### PR TITLE
feat: use anchore version retrieved by secure APIs

### PR DESCRIPTION
**Pull request by ProvoK**

### CODE REVIEW

File: inline_scan.sh [hunk 1]
```diff
@@ -8,8 +8,8 @@ set -eou pipefail
 
 # If using a locally built stateless CI container, export SYSDIG_CI_IMAGE=<image_name>.
 # This will override the image name from Dockerhub.
-INLINE_SCAN_IMAGE="${SYSDIG_CI_IMAGE:-docker.io/anchore/inline-scan:v0.6.1}"
 DOCKER_NAME="${RANDOM:-temp}-inline-anchore-engine"
+INLINE_SCAN_IMAGE="${INLINE_SCAN_IMAGE:-}"
```
Comment (ProvoK) [inline inline_scan.sh:12]: I'm thinking to use an environment variable (and not a flag) for two reasons:

1) There are already plenty of flags
2) This customisation should be done as last resort or for testing purposes

File: inline_scan.sh [hunk 2]
```diff
@@ -249,14 +249,27 @@ get_and_validate_images() {
...
 prepare_inline_container() {
-    # Check if env var is overriding which inline-scan image to utilize.
-    if [[ -z "${SYSDIG_CI_IMAGE-docker.io/anchore/inline-scan:v0.6.1}" ]]; then
-        printf '\n%s\n' "Pulling ${INLINE_SCAN_IMAGE}"
-        docker pull "${INLINE_SCAN_IMAGE}"
+    # Retrieve dynamically from secure the Anchore version for compatibility reasons
+    if [[ -z "$INLINE_SCAN_IMAGE" ]]; then
+        printf 'Retrieving remote Anchore version from Sysdig Secure APIs\n'
+        SCANNING_ANCHORE_STATUS=$(curl -sSkf  -H "Authorization: Bearer ${SYSDIG_API_TOKEN}" "${SYSDIG_SCANNING_URL%%/}/anchore/status")
+        INLINE_SCAN_IMAGE_VERSION=$(echo ${SCANNING_ANCHORE_STATUS} | grep -o -E '"version":[ \t]?".*"' | awk -F ":" '{print $2}' | awk -F '"' '{print $2}')
```
Comment (luca-sd) [inline inline_scan.sh:256]: such an intuitive language to parse jsons 😉, nice idea using `-o`
the only thing I can think of is that maybe we want to bail early if `curl` had any issues (i.e. `$?` is nonzero)

**Review by luca-sd — APPROVED** [review]
lgtm

File: inline_scan.sh (hunk 2 above)
Comment (ProvoK) [reply inline_scan.sh:256]: afaik, any command failure should exits immediately the script because of `set -eou pipefail`.

Comment (luca-sd) [reply inline_scan.sh:256]: Thanks, I missed that :)

**Pull request merged**